R/gitbook.R

```diff
@@ -138,10 +138,16 @@ gitbook_page = function(
   }
 
   # you can set the edit setting in either _bookdown.yml or _output.yml
-  if (is.list(setting <- edit_setting(config$edit))) config$edit = setting
+  if (is.list(setting <- source_link_setting(config$edit, type="edit")))
+    config$edit = setting
   if (length(rmd_cur) && is.list(config$edit))
     config$edit$link = sprintf(config$edit$link, rmd_cur)
 
+  if (is.list(setting <- source_link_setting(config$history, type="history")))
+    config$history = setting
+  if (length(rmd_cur) && is.list(config$history))
+    config$history$link = sprintf(config$history$link, rmd_cur)
```

Comment:
I think the above two blocks of code can also be collapsed into a for loop. They look too similar.

```r
for (type in c('edit', 'history')) {
  ...
}
```